system/lib/wasmfs/file.h

```diff
@@ -20,6 +20,43 @@
 
 namespace wasmfs {
 
+class path_string {
+  std::string path;
+
+public:
+  path_string() = default;
+  path_string(const std::string& p) : path(p) {}
+  path_string(const char* p) : path(p) {}
+  const std::string& str() const { return path; }
+};
+
+inline bool operator==(const path_string& x, const path_string& y) {
+#ifdef WASMFS_CASE_INSENSITIVE
+  auto&& a = x.str();
+  auto&& b = y.str();
+  return a.size() == b.size() &&
+         strncasecmp(a.c_str(), b.c_str(), b.size()) == 0;
+#else
+  return x.str() == y.str();
+#endif
+}
+
+inline bool operator!=(const path_string& x, const path_string& y) {
+  return !(x == y);
+}
+
+inline bool operator<(const path_string& x, const path_string& y) {
+#ifdef WASMFS_CASE_INSENSITIVE
+  return strcasecmp(x.str().c_str(), y.str().c_str()) < 0;
+#else
+  return x.str() < y.str();
+#endif
+}
+
+inline bool operator>(const path_string& x, const path_string& y) {
+  return y < x;
+}
```

Comment:
Do we need this operator? I would have thought that `operator<` would be enough for the dcache map.

Reply:
I added it to mirror `operator<`. But I agree, we don't need it.